src/tasks/change-plan-task.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseTask } from "./base-task";
+import { Stripe } from "stripe";
+import { DatabaseTransactionHandler, Member } from "graasp";
+import { FastifyLoggerInstance } from "fastify";
+import { PayementFailed, PlanNotFound, SubscriptionNotFound } from "../util/errors";
+import { CustomerExtra } from "../interfaces/customer-extra";
+import { Plan } from "../interfaces/plan";
+
+export class ChangePlanTask extends BaseTask<Plan> {
+  get name(): string {
+    return ChangePlanTask.name;
+  }
+
+  constructor(member: Member<CustomerExtra>, planId: string, stripe: Stripe) {
+    super(member, null, stripe);
+    this.targetId = planId;
+  }
+
+  async run(handler: DatabaseTransactionHandler, log: FastifyLoggerInstance): Promise<void> {
+    this.status = "RUNNING";
+
+    const {
+      extra: { subscriptionId },
+    } = this.actor;
+
+    const plan = await this.stripe.prices.retrieve(this.targetId, { expand: ["product"] });
+    if (!plan) throw new PlanNotFound(this.targetId);
+
+    const subscription = await this.stripe.subscriptions.retrieve(subscriptionId);
+    if (!subscription) throw new SubscriptionNotFound(this.targetId);
+
+    await this.stripe.subscriptions
+      .update(subscriptionId, {
+        billing_cycle_anchor: "now",
+        payment_behavior: "error_if_incomplete",
+        proration_behavior: "always_invoice",
+        items: [{ id: subscription.items.data[0].id, price: this.targetId }],
```

Comment:
Can you explain what the `price` is in? Is it an integer? We should avoid using floats for maths, but since Stripe is doing the heavy lifting, please add comments here for clarity.